### PR TITLE
Curation tool to update index

### PR DESCRIPTION
**Pull request by UlysseFG**
Based off the derived_from PR in order to get access to it to parse derived_samples.

### CODE REVIEW

**Review by JMarce01 — APPROVED** [review]

File: backend/fms_core/management/commands/_update_index.py
Comment (nafiz1001) [inline backend/fms_core/management/commands/_update_index.py:1]: That is a pretty deep nest.

File: backend/fms_core/management/commands/_update_index.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+COMMENT = "comment"                                     # A comment to be stored in the logs. Optional.
+DERIVED_SAMPLE_IDENTIFIER = "derived_sample_identifier" # A dictionary that contains the fields required to uniquely identify the targeted derived sample.
+INDEX_OLD = "index_old"                                 # The old value of the library's index (used for validation). Optional. Validation skipped if empty.
+INDEX_NEW = "index_new"                                 # The new value of the library's index.
+USER_ID = "requester_user_id"                           # The user id of the person requesting the curation. Optional. If left empty, uses biobankadmin id.
+
+# Curation params template
+# { CURATION_INDEX: 1,
+#   ACTION: "update_index",
+#   COMMENT: "Dr. No asked a libray index to be changed from [PATATE] to [CAROTTE].",
```
Comment (nafiz1001) [inline backend/fms_core/management/commands/_update_index.py:21]: 😆

File: backend/fms_core/management/commands/_update_index.py
Comment (nafiz1001) [reply backend/fms_core/management/commands/_update_index.py:1]: A common pattern I noticed is that for each error you...
- `log.error`
- `error_found = True`
There should be a way of flattening the code.

Comment (UlysseFG) [reply backend/fms_core/management/commands/_update_index.py:1]: Error tracking is rarely pretty. I'll try to find a way to improve this down the line.

**Pull request merged**